integration/kotlinx-coroutines-jdk8/src/future/Future.kt

```diff
@@ -159,10 +165,14 @@ public suspend fun <T> CompletionStage<T>.await(): T {
     return suspendCancellableCoroutine { cont: CancellableContinuation<T> ->
         val consumer = ContinuationConsumer(cont)
         whenComplete(consumer)
-        cont.invokeOnCancellation {
-            // mayInterruptIfRunning is not used
-            (this as? CompletableFuture<T>)?.cancel(false)
-            consumer.cont = null // shall clear reference to continuation to aid GC
+        if (cont.isActive) {
```

Comment:
Not a big fan of optimizations that work 0.01% of the time (when future completes after `isDone` check but after `whenComplete`): hard to test and reason about.
I may be too opinionated here tho

Reply:
Yep. It is indeed not needed anymore, where we (again) have fast-path via `future.isDone` here.